### PR TITLE
Remove AIP-44 from taskinstance

### PR DESCRIPTION
**Pull request by potiuk**
Part of https://github.com/apache/airflow/issues/44436

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
I want to make this one pass - and then - separately - do refactoring with merging those changes in task instance.

File: airflow/models/taskinstance.py [hunk 1]
```diff
@@ -2558,12 +2508,6 @@ def _check_and_change_state_before_execution(
 
         if isinstance(task_instance, TaskInstance):
```
Comment (jscheffl) [inline airflow/models/taskinstance.py]: Is that `if` still needed if there is no else?

File: airflow/sensors/base.py [hunk 2]
```diff
@@ -230,7 +256,7 @@ def execute(self, context: Context) -> Any:
                         TaskReschedule.run_id == ti.run_id,
                         TaskReschedule.map_index == ti.map_index,
                         TaskReschedule.try_number == first_try_number,
-                    )
+                        )
```
Comment (jscheffl) [inline airflow/sensors/base.py]: This looks like an "whitespace-uuups"?

File: providers/src/airflow/providers/amazon/aws/transfers/google_api_to_s3.py [hunk 3]
```diff
@@ -175,19 +174,15 @@ def _load_data_to_s3(self, data: dict) -> None:
             replace=self.s3_overwrite,
         )
 
-    def _update_google_api_endpoint_params_via_xcom(
-        self, task_instance: TaskInstance | TaskInstancePydantic
-    ) -> None:
+    def _update_google_api_endpoint_params_via_xcom(self, task_instance: TaskInstance) -> None:
```
Comment (jscheffl) [inline providers/src/airflow/providers/amazon/aws/transfers/google_api_to_s3.py:177]: As you change the signature in Google provider - will this not break compatibility of the provider with Airflow 2.10? Here I only see Type checking code... mhm, so functional should be working.

**Review by jscheffl — APPROVED** [review]
3 small santity check questions, else looks good in my view as a first pass cleaning.

File: airflow/models/taskinstance.py (hunk 1 above)
Comment (potiuk) [reply airflow/models/taskinstance.py]: Good point . No.

File: airflow/sensors/base.py (hunk 2 above)
Comment (potiuk) [reply airflow/sensors/base.py]: uuups..ruff fixed it.

File: providers/src/airflow/providers/amazon/aws/transfers/google_api_to_s3.py (hunk 3 above)
Comment (potiuk) [reply providers/src/airflow/providers/amazon/aws/transfers/google_api_to_s3.py:177]: Yeah. It's just non-functional change.

File: airflow/models/taskinstance.py [hunk 4]
```diff
@@ -1557,33 +1531,16 @@ def _update_rtif(ti, rendered_fields, session: Session = NEW_SESSION):
...
-            task_id=ti.task_id,
-            map_index=ti.map_index,
-            session=session,
-        )
-        if TYPE_CHECKING:
-            assert orm_ti
-        ti, pydantic_ti = orm_ti, ti
-        _set_ti_attrs(ti, pydantic_ti)
-        ti.task = pydantic_ti.task
+def _coalesce_to_orm_ti(*, ti: TaskInstance, session: Session):
```
Comment (potiuk) [inline airflow/models/taskinstance.py]: We will be able to remove it .. but in a follow up PR.

Comment (potiuk) [reply airflow/models/taskinstance.py]: Actually ... I can do it even now.

**Pull request merged**